### PR TITLE
Make remove content optional during account deletion (fixes #1617)

### LINKED ISSUE #1617: Make removing content optional upon account deletion
### Is your proposal related to a problem?

When deleting an account it would be nice if the user could decide wether or not their old posts should be removed.
See context: https://lemmy.ml/post/66254/comment/56422

### Describe the solution you'd like

Just a dialog with a choice.

### PR DESCRIPTION
**Pull request by Nutomic**
It should be possible to delete a user account without wiping all posts and comments. This can also reduce server load. 

Requires a new lemmy-js-client version for api tests.

### CODE REVIEW

File: crates/api_crud/src/user/delete.rs [hunk 1]
```diff
@@ -26,8 +27,20 @@ pub async fn delete_account(
...
+    purge_user_account(
+      local_user_view.person.id,
+      &mut context.pool(),
+      &context.settings(),
+      context.client(),
+    )
+    .await?;
+  } else {
+    Person::delete_account(&mut context.pool(), local_user_view.person.id).await?;
+  }
```
Comment (Nutomic) [inline crates/api_crud/src/user/delete.rs:40]: This was called from the federation code for some reason.

**Comment by dessalines** [comment]
Mmmk, I'll start an RC for that.

**Comment by dessalines** [comment]
Okay the new lemmy-js-client is `0.19.0-rc.2` . CI is a little backed up at the moment tho.

**Comment by Nutomic** [comment]
Updated. Locally Im still getting the same test failure, lets see if it works in ci.

**Pull request merged**